### PR TITLE
AEA-1288 Bump protocol specification id if needed

### LINKED ISSUE #2244: Generate protocols script needs to take account of protocol spec id
**Is your feature request related to a problem? Please describe.**
The generate protocols script does not care about protocol spec id at the moment.

**Describe the solution you'd like**
The script needs to check if the protocol spec id needs to be bumped; if so, bump it and generate.

### LINKED ISSUE #2247: Improve package checking scripts
**Is your feature request related to a problem? Please describe.**
Packages (agents) with missing description or wrong fingerprint info are passing. E.g. this should have been detected: https://github.com/fetchai/agents-aea/pull/2248/files

**Describe the solution you'd like**
Fix.

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Bump protocol specification id if needed.

## Fixes

Fix #2244

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: scripts/generate_all_protocols.py [hunk 1]
```diff
@@ -387,6 +468,7 @@ def main() -> None:
             log("=" * 100)
             log(f"Processing protocol at path {package_path}")
             _process_packages_protocol(package_path)
+            _bump_protocol_specification_id_if_needed(package_path)
```
Comment (DavidMinarsch) [inline scripts/generate_all_protocols.py:471]: Wrong order, bumping spec must happen before anything is processed.

File: scripts/generate_all_protocols.py [hunk 2]
```diff
@@ -366,6 +375,78 @@ def _process_test_protocol(specification: Path, package_path: Path) -> None:
...
+    old_specification_content = _get_protocol_specification_from_readme(
+        downloaded_package_directory
+    )
+    old_configuration: ProtocolConfig = cast(
+        ProtocolConfig,
+        load_component_configuration(ComponentType.PROTOCOL, temp_directory),
+    )
+
+    # if different, bump protocol specification version, else don't.
+    version_is_newer = (
```
Comment (DavidMinarsch) [inline scripts/generate_all_protocols.py:437]:
```suggestion
    public_id_version_is_newer = (
```

File: scripts/generate_all_protocols.py [hunk 3]
```diff
@@ -366,6 +375,78 @@ def _process_test_protocol(specification: Path, package_path: Path) -> None:
...
+    filepath = download_file(file_url, destination_path)
+    extract(filepath, destination_path)
+
+
+def _bump_protocol_specification_id(
+    package_path: Path, configuration: ProtocolConfig
+) -> None:
+    """Bump spec id version."""
+    spec_id: PublicId = configuration.protocol_specification_id  # type: ignore
+    log(f"Bumping protocol specification id '{spec_id}'")
```
Comment (DavidMinarsch) [inline scripts/generate_all_protocols.py:392]: print from... to ...

File: scripts/generate_all_protocols.py [hunk 4]
```diff
@@ -366,6 +375,82 @@ def _process_test_protocol(specification: Path, package_path: Path) -> None:
...
+    # if different, bump protocol specification version, else don't.
+    public_id_version_is_newer = (
+        old_configuration.public_id.package_version  # type: ignore
+        <= configuration.public_id.package_version
+    )
+    content_is_different = current_specification_content != old_specification_content
+    if public_id_version_is_newer and content_is_different:
+        log(
+            f"Bumping protocol specification id from '{configuration.protocol_specification_id}' to '{old_configuration.protocol_specification_id}'"
+        )
```
Comment (DavidMinarsch) [inline scripts/generate_all_protocols.py:446]: Wrong way round. ...

**Review by solarw — APPROVED** [review]
LGTM

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**